src/app/routes/registration/utils/precondition-check.ts

```diff
@@ -0,0 +1,63 @@
+import { Claim, PreconditionType } from 'iam-client-lib';
+import { RolePreconditionType } from '../models/role-precondition-type.enum';
+
+export interface EnrolmentPrecondition {
+  type: string,
+  conditions: string[];
+}
+
+export interface PreconditionCheck {
+  namespace: string;
+  status: RolePreconditionType;
+}
+
+export const preconditionCheck = (preconditionList: EnrolmentPrecondition[], roleClaimList: Claim[]): [boolean, PreconditionCheck[]] => {
+  let retVal = true;
+  let rolePreconditionList: PreconditionCheck[];
+  if (preconditionList && preconditionList.length) {
+    for (const precondition of preconditionList) {
+      switch (precondition.type) {
+        case PreconditionType.Role:
+          // Check for Role Conditions
+          rolePreconditionList = [];
+
+          const conditions = precondition.conditions;
+          if (conditions) {
+            for (const roleCondition of conditions) {
+              const status = getRoleConditionStatus(roleCondition, roleClaimList);
+              rolePreconditionList.push({
+                namespace: roleCondition,
+                status
+              });
+
+              if (status !== RolePreconditionType.SYNCED) {
+                retVal = false;
+              }
+            }
+          }
+          break;
+      }
+    }
+  }
+
+  return [retVal, rolePreconditionList];
+};
+
+const getRoleConditionStatus = (namespace: string, roleList) => {
+  let status = RolePreconditionType.PENDING;
+  // Check if namespace exists in synced DID Doc Roles
+  for (const roleObj of roleList) {
+    if (roleObj.claimType === namespace) {
+      if (roleObj.isAccepted) {
+        if (roleObj.isSynced) {
+          status = RolePreconditionType.SYNCED;
+        } else {
+          status = RolePreconditionType.APPROVED;
+        }
+      }
+      break;
+    }
+  }
```

Comment:
We also need a check here about `on-chain` vs `off-chain`. Because an on-chain role would be not be "synced" but it doesn't need to be synced.

Actually, testing locally I ran into this issue. I requested (and was issued) an on-chain role only and SB told me to sync it, which shouldn't be the case.
![image](https://user-images.githubusercontent.com/4407464/142678628-5d09f5bd-d1ae-4219-9bec-e9ae73f83f5d.png)

Reply:
Ok, so the problem is with displaying message about `Pending Sync to DID Document` right? It should be not possible to sync if there is an on-chain option selected. What if there is on-chain and off-chain selected? The same case -> no need to sync?
I'm not sure if the problem lays in this selected part of code, because it have nothing to do with displaying status of an enrollment.

Reply:
Hmm, yes, the issue might not be in this code. I was thinking that it might be because a role that is `on-chain` only might not have `isSynced==true` but still wouldn't need to be synced. But the issue could definitely be elsewhere.

As far as the logic, an `on-chain` credential does not need to be synced, while an 'off-chain` credential does need to be synced. So:
- If user has on-chain credential: nothing needs to be synced
- If user has off-chain credential: off-chain credential needs to be synced
- If user has both on-chain and off-chain: only off-chain needs to be synced